File: rest-service/src/main/java/mb/lib/queue/model/FailedJobResponse.java
package mb.lib.queue.model;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonGetter;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonSetter;
import mb.lib.queue.consts.JsonKeys;

@JsonIgnoreProperties(ignoreUnknown = true)
public interface FailedJobResponse<T extends FailedJob<?>>
{
  @JsonGetter(JsonKeys.FailedJobs)
  List<T> getFailedJobs();

  @JsonSetter(JsonKeys.FailedJobs)
  FailedJobResponse<T> setFailedJobs(List<T> failedJobs);
}

File: rest-service/src/main/java/mb/lib/queue/model/FailedJobResponse.kt
package mb.lib.queue.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonValue
import com.fasterxml.jackson.databind.JsonNode

@JsonIgnoreProperties(ignoreUnknown = true)
interface FailedJobResponse<T: FailedJob<*>>
{
  val failedJobs: List<T>

  @JsonValue
  fun toJSON(): JsonNode
}
